include/pgduckdb/pgduckdb_utils.hpp

```diff
@@ -19,4 +23,44 @@ TokenizeString(char *str, const char delimiter) {
 	return v;
 };
 
+template <typename T, typename FuncType, typename... FuncArgs>
+std::optional<T>
+PostgresFunctionGuard(FuncType postgres_function, FuncArgs... args) {
+	T return_value;
+	bool error = false;
```

Comment:
Needs to be marked `volatile`

Reply:
Actually it seems this is probably not necessary. I was thinking about the problem described here, but that does not apply since `error` is not changed in the PG_TRY section: https://github.com/postgres/postgres/blob/master/src/include/utils/elog.h#L369-L374